pywbem/_logging.py

```diff
@@ -14,371 +14,303 @@
-                      log_detail_level=DEFAULT_LOG_DETAIL_LEVEL):
-        """
-        Create the logger defined by the input parameters and place the result
-        in a class level dictionary in this class.
+    2018-03-17 11:39:09,877-pywbem.api.1-32073-Connection:1-32073 WBEMConnection(url='http://localhost', creds=None, default_namespace='root/cimv2')
+    2018-03-17 11:41:13,176-pywbem.api.1-32073-Request:1-32073 EnumerateClasses(ClassName=None, DeepInheritance=None, IncludeClassOrigin=None, IncludeQualifiers=None, LocalOnly=None, namespace=None)
+    2018-03-17 11:41:13,635-pywbem.api.1-32073-Return:1-32073 EnumerateClasses(list of CIMClass; count=103)
 
-        This function can be used to set up all of the named loggers used by
-        pywbem.
+"""   # noqa: E501
+# pylint: enable=line-too-long
 
-        Parameters:
+import logging
+
+# NOTE: cannot use from pywbem import WBEMConnection because it is circular
+# import
+import pywbem
 
-          log_component (:term:`string`):
-            The name of the logger. It must be one of the
-            names defined in :data:`~pywbem._logging.LOG_COMPONENTS`.
-            Used to create the logger name by prepending with the logger name
-            prefix ``pywbem.``.
+__all__ = ['define_loggers_from_string',
+           'LOG_API_CALLS_NAME', 'LOG_HTTP_NAME', 'LOG_COMPONENTS',
+           'LOG_DESTINATIONS', 'DEFAULT_LOG_DETAIL_LEVEL',
+           'DEFAULT_API_LOG_DEFAULT_LEVEL', 'DEFAULT_HTTP_LOG_DETAIL_LEVEL',
+           'DEFAULT_LOG_DESTINATION', 'LOG_DETAIL_LEVELS']
 
-          log_dest (:term:`string`):
-            String defining the destination for this log. It must be one of the
-            destinations defined in :data:`~pywbem._logging.LOG_DESTINATIONS`
-            or `None`. If the value is the string ``none``, the null logger is
-            created.
+#: Name of logger for logging user-issued calls to
+#: :class:`~pywbem.WBEMConnection` methods that drive WBEM operations.
+LOG_API_CALLS_NAME = 'pywbem.api'
 
-          log_filename (:term:`string`):
-            Filename to use as logging file if the log destination is `file`.
-            Ignored if log destination is not `file`. If value is `None` and
-            this is a ``file`` log, ValueError is raised.
+#: Name of logger for HTTP requests and responses between the pywbem client and
+#: WBEM servers.
+LOG_HTTP_NAME = 'pywbem.http'
 
-          log_detail_level (:term:`string`):
-            String defining the level of detail for log output. This is
-            optional. The default is defined in
-            :data:`~pywbem._logging.DEFAULT_LOG_DETAIL_LEVEL`.
+#: List of the logger names that the logging configuration helper functions
+#: recognize. This allows setting up either logger individually or both to the
+#: same destination.
+LOG_COMPONENTS = ['api', 'http', 'all']
 
-        Raises:
+#: List of log destinations that the logging configuration helper functions
+#: recognize.
```

Comment:
The log destination values should be described, along with additional requirements, such as that 'file' requires specifying the file name.

Reply:
They are but in the wrong place, the _logging module doc string.

DISCUSSION: I want to do this only once so need to sort out where it goes to be generally referencable.  That is complicated by the fact that we ended up with 3 configure methods.

Reply:
I have put them with the variable and am now trying to do links from other places to that variable. Will not have ready for telecon this morning.